File: app/models/product.rb
# frozen_string_literal: true

# Model for products
class Product < ApplicationRecord
  validates :title, :author, :release_date, :description, :category, presence: true
  validates :title, uniqueness: { scope: :author,
                                  message: 'Le titre existe déjà pour cet auteur' }
  belongs_to :section
  has_many :likes, dependent: :destroy
  has_many :comments, dependent: :destroy
end

File: client/src/components/Product.js
import React from 'react';
import Like from './Like';

const Product = ({product}) =>
    <tr className="single-product" key={product.id}>
        <td>{product.title}</td>
        <td>{product.author}</td>
        <td><Like product={product} /></td>
    </tr>
export default Product;
